frame/nomination-pools/src/lib.rs

```diff
@@ -990,16 +997,24 @@ pub mod pallet {
 		}
 	}
 
+	/// Events of this pallet.
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
 	pub enum Event<T: Config> {
+		/// A pool has been created.
 		Created { depositor: T::AccountId, pool_id: PoolId },
-		Joined { delegator: T::AccountId, pool_id: PoolId, bonded: BalanceOf<T> },
+		/// A delegator has became bonded in a pool.
+		Bonded { delegator: T::AccountId, pool_id: PoolId, bonded: BalanceOf<T>, joined: bool },
```

Comment:
What about two separate events `Bonded` and `BondedExtra`? I think its more clear and will mean the events take 1 byte less space

Reply:
I prefer a single event with `joined: bool`, but happy to go either way.